client/src/main/java/org/apache/falcon/ExtensionHandler.java

```diff
@@ -185,8 +185,6 @@ private static String createStagePath(String extensionName, String jobName) {
                 for (File innerFile : files) {
                     if (innerFile.isFile()) {
                         urls.add(innerFile.toURI().toURL());
```

Comment:
Yes. There are occasional build failures too for recursive calling. I will add it to the documentation.